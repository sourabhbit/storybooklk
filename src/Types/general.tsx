export enum ComponentSizeENUM {
	small = 'small',
	medium = 'medium',
	large = 'large',
}

export enum TypographyENUM {
	sans = 'var(--font-sans)',
	serif = 'var(--font-serif)',
}

export enum ThemeENUM {
	primary = 'primary',
	secondary = 'secondary',
}
