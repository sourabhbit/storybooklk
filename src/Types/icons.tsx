export enum iconsENUM {
	addToCart = 'addToCart',
	addedToCart = 'addedToCart',
	antiScratch = 'antiScratch',
	backArrow = 'backArrow',
	bin = 'bin',
	cross = 'cross',
	crossWithSpace = 'crossWithSpace',
	edit = 'edit',
	emptyCart = 'emptyCart',
	fullWidth = 'fullWidth',
	gift = 'gift',
	glasses = 'glasses',
	globe = 'globe',
	gridList = 'gridList',
	phone = 'phone',
	share = 'share',
	star = 'star',
	starFilled = 'starFilled',
	tick = 'tick',
	user = 'user',
	group = 'group',
}
