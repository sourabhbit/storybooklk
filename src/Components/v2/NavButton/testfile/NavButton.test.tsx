import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { ComponentSizeENUM, TypographyENUM } from '../../../../Types/general'
import Navbutton from '../index'

const props = {
	id: 'nav-button-id',
	text: 'Your Picks',
	componentSize: ComponentSizeENUM.small,
	font: TypographyENUM.serif,
	selectedNav: 1,
	index: 0,
	onClick: jest.fn(),
}

describe('<Navbutton />', () => {
	let _asFragment: any = null

	beforeEach(() => {
		const { asFragment } = render(<Navbutton {...props} />)
		_asFragment = asFragment
	})

	afterAll(() => {
		_asFragment = null
	})

	it('Should render component', () => {
		const container = screen.getByTestId('nav-button')
		expect(container).toBeInTheDocument()
	})

	it('Should text component', () => {
		const container = screen.getByTestId('nav-text')
		expect(container).toBeInTheDocument()
	})

	it('Should take a snapshot', () => {
		expect(_asFragment()).toMatchSnapshot()
	})
})
