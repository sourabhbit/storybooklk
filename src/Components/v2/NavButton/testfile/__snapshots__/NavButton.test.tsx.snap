// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbutton /> Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL hpipGf"
    data-testid="nav-button"
    id="nav-button-id"
  >
    <span
      class="sc-gsDKAQ izrzXn"
      data-testid="nav-text"
    >
      Your Picks
    </span>
  </div>
</DocumentFragment>
`;
